File: backup_nodejs_20250727_135816/models/topup_card.py
from flask_sqlalchemy import SQLAlchemy
from datetime import datetime
import uuid

db = SQLAlchemy()

class TopupCard(db.Model):
    __tablename__ = 'topup_cards'
    id = db.Column(db.String(36), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = db.Column(db.String(36), db.ForeignKey('users.id'), nullable=False)
    card_type = db.Column(db.String(50), nullable=False)
    amount = db.Column(db.Float, nullable=False)
    serial = db.Column(db.String(100), nullable=False)
    code = db.Column(db.String(100), nullable=False)
    status = db.Column(db.String(20), default='Đang duyệt')
    created_at = db.Column(db.DateTime, default=datetime.utcnow)

    def to_dict(self):
        return {
            'id': self.id,
            'user_id': self.user_id,
            'card_type': self.card_type,
            'amount': self.amount,
            'serial': self.serial,
            'code': self.code,
            'status': self.status,
            'created_at': self.created_at.isoformat()
        } 

File: backup_nodejs_20250727_135816/models/TopupCard.js
const mongoose = require('mongoose');
const TopupCardSchema = new mongoose.Schema({
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  cardType: { type: String, required: true },
  amount: { type: Number, required: true },
  serial: { type: String, required: true },
  code: { type: String, required: true },
  status: { type: String, enum: ['Đang duyệt', 'Thành công', 'Thất bại'], default: 'Đang duyệt' },
  createdAt: { type: Date, default: Date.now }
});
module.exports = mongoose.model('TopupCard', TopupCardSchema); 